File: Go/recursion/towerOfHanoi.go
package main
import "fmt"

func towerOfHanoi(n int, source string, helper string, destination string)  {
    if n == 0{
        return
    } 
    towerOfHanoi(n-1, source, destination, helper)
    fmt.Printf("Disk %d from rod %s to rod %s\n", n, source, destination)
    towerOfHanoi(n-1, helper, source, destination)
}
func main() {
   towerOfHanoi(3, "A","B","C") 
}

File: Java/recursion/towerOfHanoi.java
class TowerOfHanoi {
    public static void towerOfHanoi(int n, String source, String helper, String destination) {
        if (n == 0){
            return;
        }
        towerOfHanoi(n-1, source, destination, helper);
        System.out.format("Disk %d from rod %s to rod %s\n", n, source, destination);
        towerOfHanoi(n-1, helper, source, destination);
    }
    public static void main(String []args) {
        towerOfHanoi(3,"A","B","C"); 
    }
}
